state-transition/core/state_processor.go

```diff
@@ -375,36 +374,19 @@ func (sp *StateProcessor[
 ]) processEpoch(
 	st BeaconStateT,
 ) (transition.ValidatorUpdates, error) {
-	slot, err := st.GetSlot()
-	if err != nil {
+	if err := sp.processRewardsAndPenalties(st); err != nil {
 		return nil, err
 	}
-
-	switch {
-	case sp.cs.DepositEth1ChainID() == spec.BartioChainID:
-		if err = sp.hollowProcessRewardsAndPenalties(st); err != nil {
-			return nil, err
-		}
-	case sp.cs.DepositEth1ChainID() == spec.BoonetEth1ChainID &&
-		slot < math.U64(spec.BoonetFork3Height):
-		// We cannot simply drop hollowProcessRewardsAndPenalties because
-		// appHash accounts for the list of operations carried out
-		// over the state even if the operations does not affect the state
-		// (rewards and penalties are always zero at this stage of beaconKit)
-		if err = sp.hollowProcessRewardsAndPenalties(st); err != nil {
-			return nil, err
-		}
-	default:
-		// no real need to perform hollowProcessRewardsAndPenalties
+	if err := sp.processRegistryUpdates(st); err != nil {
+		return nil, err
 	}
```

Comment:
moved into `processRewardsAndPenalties` to have fork switches less in the way of mostly read code